### PR TITLE
Subscription Management: Add and integrate useSiteSubscriptionDetailsQuery.

### PR DESCRIPTION
**Pull request by yansern**
Related to # (don't have an issue created for this yet)

## Proposed Changes

* Add useSiteSubscriptionDetailsQuery.
* Integrate it to the existing individual subscription management page.

## Testing Instructions

* Test as both logged-in (return next() on client/server/pages/index.js) and external user (subkey).
* Go to `http://calypso.localhost:3000/subscriptions/site/{site_id_here}`.
* It should display with actual data (except for subscriber count).

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: packages/data-stores/src/reader/types/index.ts [hunk 1]
```diff
@@ -131,3 +131,23 @@ export type PendingPostSubscriptionsResult = {
...
+
+export type SiteSubscriptionDetailsAPIResponse = {
+	ID: string;
+	blog_ID: string;
+	name: string;
+	URL: string;
+	site_icon: string;
+	date_subscribed: Date;
+	delivery_methods: SiteSubscriptionDeliveryMethods;
+};
```
Comment (yansern) [inline packages/data-stores/src/reader/types/index.ts:153]: For now, both types look the same, this may change later as this is just first quick iteration.

File: client/landing/subscriptions/components/site-subscription-page/site-subscription-page.tsx [hunk 2]
```diff
@@ -168,18 +156,18 @@ const SiteSubscriptionPage = () => {
 						<>
 							<SiteSubscriptionSettings
 								blogId={ blogId }
-								notifyMeOfNewPosts={ notifyMeOfNewPosts }
-								emailMeNewPosts={ emailMeNewPosts }
-								deliveryFrequency={ deliveryFrequency }
-								emailMeNewComments={ emailMeNewComments }
+								notifyMeOfNewPosts={ data.delivery_methods?.notification?.send_posts }
```
Comment (yansern) [inline client/landing/subscriptions/components/site-subscription-page/site-subscription-page.tsx:159]: Take note that API is returning `notifications` which will be fixed in another patch. And `notification` object doesn't exist for external user, so `?` optional operator is still required.

File: packages/data-stores/src/reader/queries/use-site-subscription-details-query.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+const useSiteSubscriptionDetailsQuery = ( siteId: string ) => {
+	const { isLoggedIn } = useIsLoggedIn();
+	const enabled = useIsQueryEnabled();
+	const cacheKey = useCacheKey( [ 'read', 'site-subscription-details', siteId ] );
+	return useQuery< SiteSubscriptionDetails >(
+		cacheKey,
+		async () => {
+			const subscriptionDetails = await callApi< SiteSubscriptionDetailsAPIResponse >( {
+				path: '/read/sites/' + siteId + '/subscription-details',
```
Comment (ivan-ottinger) [inline packages/data-stores/src/reader/queries/use-site-subscription-details-query.ts:14]: I prefer using template literals for readability, but that's just my personal preference. 🙂

File: client/landing/subscriptions/components/site-subscription-page/site-subscription-page.tsx [hunk 4]
```diff
@@ -111,12 +92,22 @@ const SiteSubscriptionPage = () => {
...
-	}, [ siteSubscribed, unsubscribeError, subscribeError, subscribing, unsubscribing ] );
+	}, [
+		siteSubscribed,
+		unsubscribeError,
+		subscribeError,
+		subscribing,
+		unsubscribing,
+		data,
+		translate,
+		blogId,
```
Comment (ivan-ottinger) [inline client/landing/subscriptions/components/site-subscription-page/site-subscription-page.tsx:108]: Thanks for adding `blogId` and `translate` to the dependency array. I missed that in my previous PR.

**Review by ivan-ottinger — APPROVED** [review]
The code looks good and works as expected. 👍🏼👍🏼

- works well for both logged-in and external users
- I did not notice any TS / JS errors nor warnings
- I did not notice any regressions

A couple things we will need to handle in subsequent PR(s):

- looks like the API isn't returning correct value for `send_comments` setting

Example:

Correct response from existing "Sites" API (that returns all the sites):
Markup on 2023-05-19 at 11:02:04

Incorrect response from the "new" API (that returns just one site)
Markup on 2023-05-19 at 11:01:21

 - optimistic updates for the settings' changes
 - error handling when the user requests details about non-existent site / site they are not subscribed to; this is the current output:
 
Markup on 2023-05-19 at 10:45:11
Markup on 2023-05-19 at 10:45:51

**Review by ivan-ottinger — APPROVED** [review]
The code looks good and works as expected. 👍🏼👍🏼

- works well for both logged-in and external users
- I did not notice any TS / JS errors nor warnings
- I did not notice any regressions

A couple things we will need to handle in subsequent PR(s):

- looks like the API isn't returning correct value for `send_comments` setting

Example:

Correct response from existing "Sites" API (that returns all the sites):
Markup on 2023-05-19 at 11:02:04

Incorrect response from the "new" API (that returns just one site)
Markup on 2023-05-19 at 11:01:21

 - optimistic updates for the settings' changes
 - error handling when the user requests details about non-existent site / site they are not subscribed to; this is the current output:
 
Markup on 2023-05-19 at 10:45:11
Markup on 2023-05-19 at 10:45:51

**Comment by yansern** [comment]
Thanks. Will address others separately.
